### ISSUE TITLE
Doesn't run on Linux

### ISSUE THREAD

**Issue by MairusuPawa** [Vocaluxe/Vocaluxe#31]
Please fix.

**Comment by derivator** [comment]
@MairusuPawa Linux support has been merged and it's very playable.

**Comment by AlexanderRavenheart** [comment]
@derivator Hy. Can you please write a step by step tutorial on how to get it running on something like Ubuntu?

**Comment by stsundermann** [comment]
@AlexanderRavenheart On ubuntu 14.10 (should apply to 14.04 as well) you will need ffmpeg which is not in the official repos but can be pulled from ppa or installed from source. The same goes for mono from the packages which will fail compiling, but luckily there's a official mono ppa which pulls in the newest version which compiles fine. http://www.mono-project.com/docs/getting-started/install/linux/

Just clone the source cd into it and type make once you have all the dependencies. For ffmpeg I installed from ppa from here http://www.noobslab.com/2014/12/ffmpeg-returns-to-ubuntu-1410.html

**Comment by AlexanderRavenheart** [comment]
@stsundermann So... what are the dependencies exactly? Do I have to install all the ffmpeg related and all the mono related packages?

**Comment by stsundermann** [comment]
If I recap correctly mono-devel and ffmpeg-* should be sufficient.

**Comment by derivator** [comment]
@AlexanderRavenheart @stsundermann At the moment, gstreamer-sharp is also a dependency. There was a problem with the dlls from the Vocaluxe repository that I solved by having the makefile delete them so the ones from the gac are used (see our [previous conversation](https://github.com/derivator/Vocaluxe/commit/3a3fc10a79484ad03102070b07a7e03a54763669) about this... I had kind of forgotten about that, I only remembered how I fixed it by deleting the dlls and assumed that they're windows-specific versions. If a solution with .dll.config files as you suggested is possible, maybe that would be preferable?)

My plan is to see what changes are required to turn Vocaluxe into an installable package on GNU/Linux systems, implement these changes, and make an AUR package for Arch Linux, as that is what I use. After that I'll look into packaging for other popular distributions.

**Comment by AlexanderRavenheart** [comment]
It fails to build. Here's the output:
Warnings:

/home/alexandru/git/Vocaluxe/Vocaluxe.sln:  warning : Ignoring vcproj 'PitchTracker'.
/home/alexandru/git/Vocaluxe/Vocaluxe.sln:  warning : Failed to find project 54a48af7-515c-4edd-b395-2bffa096050a

Errors:

/home/alexandru/git/Vocaluxe/Vocaluxe.sln (default targets) ->
(Build target) ->
/home/alexandru/git/Vocaluxe/Vocaluxe/Vocaluxe.csproj (default targets) ->
/usr/lib/mono/4.5/Microsoft.CSharp.targets (CoreCompile target) ->

	Lib/Input/CGamePad.cs(19,14): error CS0234: The type or namespace name `PerformanceData' does not exist in the namespace `System.Diagnostics'. Are you missing an assembly reference?

Any thougths on how to solve this?

**Comment by stsundermann** [comment]
Did you install mono from the ppa provided above? Maybe the missing stuff is in mono-complete?

**Comment by AlexanderRavenheart** [comment]
I installed both mono-complete and mono-devel from the ppa mentioned above. Running on Ubuntu Trusty (14.04).

**Comment by derivator** [comment]
I'm getting the same build error. (It seems that that namespace is not actually used, so you can just comment the offending line, but then I still get a NotImplementedException from OpenTK on `_HandleButtons(GamePad.GetState(_GamePadIndex));`) For now, you can fix it by doing `git revert e48d71`.

@lukeIam Do we need something from `PerformanceData`? Any ideas about the NotImplementedException? Is OpenTK on linux lagging behind?

**Comment by stsundermann** [comment]
Not right now since I don't have a linux server machine to install the buildbot on.

**Comment by AlexanderRavenheart** [comment]
Ok, I deleted the Vocaluxe directory and did a fresh git clone.
After that I cd-ed into the directory and ran the command git revert e48d71.
I tried make and it failed with 16 errors regarding GstreamerSharp.

I used the Building & Installing instruction provided on https://github.com/gstreamer-sharp/gstreamer-sharp after git cloning that repository to generate the gstreamer file.

After that I tried make again on Vocaluxe and failed with the same errors as before.

/usr/lib/mono/4.5/Microsoft.CSharp.targets (CoreCompile target) ->

	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(75,35): error CS1502: The best overloaded method match for `Gst.Element.QueryPosition(ref Gst.Format, out long)' has some invalid arguments
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(75,56): error CS1620: Argument `#1' is missing `ref' modifier
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(78,56): error CS0103: The name `Constants' does not exist in the current context
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(85,30): error CS1061: Type `Gst.Element' does not contain a definition for `SeekSimple' and no extension method `SeekSimple' of type `Gst.Element' could be found. Are you missing an assembly reference?
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(96,55): error CS1061: Type `Gst.Element' does not contain a definition for `TargetState' and no extension method `TargetState' of type `Gst.Element' could be found. Are you missing an assembly reference?
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(150,31): error CS1540: Cannot access protected member `Gst.Element.AddPad(Gst.Pad)' via a qualifier of type `Gst.Bin'. The qualifier must be of type `Vocaluxe.Lib.Sound.Playback.GstreamerSharp.CGstreamerSharpAudioStream' or derived from it
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(150,31): error CS0122: `Gst.Element.AddPad(Gst.Pad)' is inaccessible due to its protection level
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(178,40): error CS1061: Type `Gst.Bus' does not contain a definition for `TimedPopFiltered' and no extension method `TimedPopFiltered' of type `Gst.Bus' could be found. Are you missing an assembly reference?
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(201,25): error CS1061: Type `Gst.Element' does not contain a definition for `TargetState' and no extension method `TargetState' of type `Gst.Element' could be found. Are you missing an assembly reference?
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(253,25): error CS1502: The best overloaded method match for `Gst.Message.ParseError(out System.Enum, out string)' has some invalid arguments
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(253,40): error CS1503: Argument `#1' cannot convert `out GLib.GException' expression to type `out System.Enum'
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(256,34): error CS0117: `Gst.MessageType' does not contain a definition for `DurationChanged'
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(278,30): error CS1502: The best overloaded method match for `Gst.Element.QueryDuration(ref Gst.Format, out long)' has some invalid arguments
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(278,51): error CS1620: Argument `#1' is missing `ref' modifier
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudioStream.cs(279,48): error CS0103: The name `Constants' does not exist in the current context
	Lib/Sound/Playback/GstreamerSharp/CGstreamerSharpAudio.cs(48,40): error CS0117: `Gst.Application' does not contain a definition for `IsInitialized'

**Comment by stsundermann** [comment]
Did you specify --prefix=/usr when running ./autogen.sh ? Else the library might end in the wrong gac

**Comment by AlexanderRavenheart** [comment]
Tried with and without the prefix option. Same result.

**Comment by stsundermann** [comment]
Did you install libgstreamer-cil-dev ? It seems like your installation is referencing the old 0.10 gstreamer bindings

**Comment by derivator** [comment]
Do you have a folder `/usr/lib/mono/gac/gstreamer-sharp` ? If yes, what's in there?

Thanks for your feedback by the way, I'll work to make building Vocaluxe on linux easier.

**Comment by AlexanderRavenheart** [comment]
@derivator 
ls /usr/lib/mono/gstreamer-sharp/
gstreamer-sharp.dll

ls /usr/lib/mono/gac/gstreamer-sharp/
0.9.2.0__4956b48baf980190  0.99.0.0__5cbf79803e97ff61

You are welcome, thank you for your efforts in bringing this wonderful piece of entertainment to the Linux world.

@stsundermann 
There is no package libgstreamer-cil-dev.
I had installed the following packages:
* libgstreamer0.10-0
* libgstreamer0.10-0-dbg
* libgstreamer0.10-cil-dev
* libgstreamer0.10-dev
* libgstreamer0.9-cil
* libgstreamer1.0-0
* libgstreamer1.0-0-dbg
* libgstreamer1.0-dev

**Comment by stsundermann** [comment]
Alright the problem is that you installed gstreamer-sharp 0.99.0 but Vocaluxe needs 1.39.1, so you need to upgrade that. This will require you to update your gstreamer installation to 1.4+

**Comment by AlexanderRavenheart** [comment]
@stsundermann 
How do I do that?

**Comment by derivator** [comment]
Installing from https://github.com/gstreamer-sharp/gstreamer-sharp should have worked. Did you maybe do just `make` instead of `make install` ?

**Comment by stsundermann** [comment]
cd gstreamer-sharp
git reset --hard # just to make sure everything is clean
git checkout master
./autogen.sh --prefix=/usr
make
sudo make install

autogen will throw an error if you don't have gstreamer 1.4+ If that happens you can either install gstreamer packages from source or try this ppa https://launchpad.net/~ddalex/+archive/ubuntu/gstreamer which has 1.4.3 for 14.04

**Comment by AlexanderRavenheart** [comment]
configure: error: Package requirements (glib-sharp-3.0 >= 2.99.4) were not met:

Requested 'glib-sharp-3.0 >= 2.99.4' but version of GLib is 2.99.2

**Comment by lukeIam** [comment]
> @lukeIam Do we need something from PerformanceData? 

no we don't -> https://github.com/Vocaluxe/Vocaluxe/pull/142

> Any ideas about the NotImplementedException? Is OpenTK on linux lagging behind?

Did you build OpenTK yourself or did you use a prebuild binary?

**Comment by derivator** [comment]
@lukeIam I'm using the one from the repository.

**Comment by lukeIam** [comment]
I've updated those in the repo (the old version did not support the Gamepad class).

Could it be that the Mono OpenTK implemention is used?
https://github.com/mono/opentk/blob/master/Source/OpenTK/Input/GamePad.cs
vs
https://github.com/opentk/opentk/blob/develop/Source/OpenTK/Input/GamePad.cs

**Comment by derivator** [comment]
@lukeIam [This exception in GamePadConfiguration.cs](https://github.com/opentk/opentk/commit/e5cd6a3b9e940207e25921e3e02b475277e66094) is the one I'm getting. Does that make sense? Which version did you update to?

**Comment by derivator** [comment]
@AlexanderRavenheart can you try with [these](https://gist.github.com/derivator/fa0bdf9995e825c3a9fb) files? The first two go into Output/ and the makefile replaces the one in the main folder. Then do `make clean` and then try building again with `make`. That should allow you to use gstreamer-sharp.dll from the Vocaluxe repository instead of building your own. Please report back if it works :)

**Comment by AlexanderRavenheart** [comment]
@derivator 
Build succeded with those files.

**Comment by lukeIam** [comment]
@derivator I used the release from their homepage: opentk-2014-07-23 should be Opentk 1.1

> This exception in GamePadConfiguration.cs is the one I'm getting. Does that make sense? 

seems that the current opentk has some problems with detecting a some sticks on the gamepad.
(I only own a x360 gamepad which is working)

The commit you send would fix the problem in my opinion, but it seems it is not included in their releases yet.

**Comment by AlexanderRavenheart** [comment]
@lukeIam 
Once it is running on my system I can try using a Hama and a Natec Genesis PV44 gamepad if you tell me how and what should I test.

**Comment by derivator** [comment]
@lukeIam, indeed, unplugging my gamepad fixed it :D Maybe we should catch those exceptions though?

**Comment by lukeIam** [comment]
@AlexanderRavenheart just plug it in and start Vocaluxe -> the first gamepad that is found will be used

@derivator 
> indeed, unplugging my gamepad fixed it :D

> Maybe we should catch those exceptions though?

I just used their docu and didn't even know that they throw such an execption :see_no_evil: 
The question is: will the gamepad be initialized if we just catch the exception.
(If not we must get sure not to use a half initialized gamepad)
An other possibility would be to build our own release (including their fix).

**Comment by AlexanderRavenheart** [comment]
@lukeIam 
It built a x64 version but my system is using x86 libraries, kernel, etc. 
How do I build a x86 version?

**Comment by derivator** [comment]
@AlexanderRavenheart I just kind of assumed everyone uses x64 these days, so I didn't plan for x86, but it should be possible to do an x86 build by just changing the platform property to x86 in the makefile. The makefiles for the native components don't explicitly specify x64, so they should build x86 assemblies if you're on x86. Let me know if there's any more problems...

**Comment by AlexanderRavenheart** [comment]
@derivator 
Please read what I wrote in my above post as I had edited in some additional details before your post went up and I would like to avoid double posting to keep this thread as clean and concise as possible.

**Comment by derivator** [comment]
@AlexanderRavenheart The problem with the videos is simply that they're not in the repository. (I suppose this should be mentioned). You can get the missing files from [here](http://sourceforge.net/projects/vocaluxe/files/Vocaluxe%200.3.x/Vocaluxe%200.3.1%20Beta.rar/download)

I don't think we depend on pulseaudio, but I haven't tested it.

**Comment by AlexanderRavenheart** [comment]
![screenshot - 28 02 2015 - 22 02 28](https://cloud.githubusercontent.com/assets/7274337/6427845/8d773b84-bf95-11e4-884b-d151f167d73a.png)

After copying the video files from the archive this is what appears.

**Comment by derivator** [comment]
Could you change the configuration from ReleaseLinux to DebugLinux in the makefile and rebuild, that should hopefully enable line-numbers for those errors...

**Comment by AlexanderRavenheart** [comment]
I changed it, ran make clean and make.
The error messages did not change, but instead of appearing as windows, they were written in the terminal window.

**Comment by derivator** [comment]
That's unfortunate. Is there anything in Error.log or Debug.log?

**Comment by AlexanderRavenheart** [comment]
Error.log:
﻿Error-Log Vocaluxe "Shining Heaven" v0.3.0 (x86) Beta (GITVERSION) 2/28/2015 10:38:50 PM
----------------------------------------

1) Cannot create directory "": Path is empty
2) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
3) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
4) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
5) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
6) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
7) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
8) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
9) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
10) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
11) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
12) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
13) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
14) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
15) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
16) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
17) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
18) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
19) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
20) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
21) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
22) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
23) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
24) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
25) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
26) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
27) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
28) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
29) Skin Ambient:Blue is missing the video "BG_Video"! Expect visual problems!
30) Error CHIDAPI.Init(): System.DllNotFoundException: libhidapi.so
  at (wrapper managed-to-native) Vocaluxe.Lib.Input.CHIDApi:hid_init ()
  at Vocaluxe.Lib.Input.CHIDApi.Init () [0x00000] in <filename unknown>:0 
31) WiiMoteLib: Can't initialize HID API
32) Please install the Visual C++ Redistributable Packages 2008!
33) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
34) Skin Default:Default is missing the video "BG_Video"! Expect visual problems!
35) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
36) Skin Default:Default is missing the video "BG_Video"! Expect visual problems!
37) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
38) Skin Default:Default is missing the video "BG_Video"! Expect visual problems!
39) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
40) Skin Default:Default is missing the video "BG_Video"! Expect visual problems!
41) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
42) Skin Default:Default is missing the video "BG_Video"! Expect visual problems!
43) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
44) Skin Default:Default is missing the video "BG_Video"! Expect visual problems!
45) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
46) Skin Default:Default is missing the video "BG_Video"! Expect visual problems!
47) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
48) Skin Default:Default is missing the video "BG_Video"! Expect visual problems!
49) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]BG_Video.mp4
50) Skin Default:Default is missing the video "BG_Video"! Expect visual problems!
51) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]IntroIn.mp4
52) Skin Ambient:Blue is missing the video "IntroIn"! Expect visual problems!
53) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]IntroMid.mp4
54) Skin Ambient:Blue is missing the video "IntroMid"! Expect visual problems!
55) Error opening video file: /home/alexandru/git/Vocaluxe/Output/Themes/Ambient/[Blue]IntroOut.mp4
56) Skin Ambient:Blue is missing the video "IntroOut"! Expect visual problems!

**Comment by derivator** [comment]
I've found the problem. (I'm stupid and made a mistake.) Move `libacinerella.so` from Output into the subfolder Output/x86 and it should work.

**Comment by AlexanderRavenheart** [comment]
Error log:
1) Cannot create directory "": Path is empty
2) Error CHIDAPI.Init(): System.DllNotFoundException: libhidapi.so
  at (wrapper managed-to-native) Vocaluxe.Lib.Input.CHIDApi:hid_init ()
  at Vocaluxe.Lib.Input.CHIDApi.Init () [0x00000] in <filename unknown>:0 
3) WiiMoteLib: Can't initialize HID API
4) Please install the Visual C++ Redistributable Packages 2008!

Terminal:
Unhandled Exception:
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeInitializationException: An exception was thrown by the type initializer for Gst.Message ---> System.EntryPointNotFoundException: gstsharp_gst_message_get_type_offset
  at (wrapper managed-to-native) Gst.Message:gstsharp_gst_message_get_type_offset ()
  at Gst.Message..cctor () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at (wrapper managed-to-native) System.Reflection.MonoCMethod:InternalInvoke (System.Reflection.MonoCMethod,object,object[],System.Exception&)
  at System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) [0x00000] in <filename unknown>:0 
  at System.Reflection.MonoCMethod.DoInvoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
  at System.Reflection.MonoCMethod.Invoke (BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
  at System.Activator.CreateInstance (System.Type type, BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes) [0x00000] in <filename unknown>:0 
  at System.Activator.CreateInstance (System.Type type, System.Object[] args, System.Object[] activationAttributes) [0x00000] in <filename unknown>:0 
  at System.Activator.CreateInstance (System.Type type, System.Object[] args) [0x00000] in <filename unknown>:0 
  at GLib.Opaque.GetOpaque (IntPtr o, System.Type type, Boolean owned) [0x00000] in <filename unknown>:0 
  at Gst.Bus.TimedPopFiltered (UInt64 timeout, MessageType types) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Lib.Sound.Playback.GstreamerSharp.CGstreamerSharpAudioStream.Open (Boolean prescan) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Lib.Sound.Playback.CPlaybackBase.Load (System.String medium, Boolean loop, Boolean prescan) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CSound.Load (System.String media, Boolean loop, Boolean prescan) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CBsound.Load (System.String soundFile, Boolean loop, Boolean prescan) [0x00000] in <filename unknown>:0 
  at VocaluxeLib.Utils.Player.CSoundPlayer.Load (System.String file, Single position, Boolean autoplay) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CBackgroundMusic._StartSong () [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CBackgroundMusic.Next () [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CBackgroundMusic.Play () [0x00000] in <filename unknown>:0 
  at Vocaluxe.Screens.CScreenLoad.OnShowFinish () [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CGraphics.InitFirstScreen () [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CDraw.MainLoop () [0x00000] in <filename unknown>:0 
  at Vocaluxe.CMainProgram._Run (System.String[] args) [0x00000] in <filename unknown>:0 
  at Vocaluxe.CMainProgram.Main (System.String[] args) [0x00000] in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeInitializationException: An exception was thrown by the type initializer for Gst.Message ---> System.EntryPointNotFoundException: gstsharp_gst_message_get_type_offset
  at (wrapper managed-to-native) Gst.Message:gstsharp_gst_message_get_type_offset ()
  at Gst.Message..cctor () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at (wrapper managed-to-native) System.Reflection.MonoCMethod:InternalInvoke (System.Reflection.MonoCMethod,object,object[],System.Exception&)
  at System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) [0x00000] in <filename unknown>:0 
  at System.Reflection.MonoCMethod.DoInvoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
  at System.Reflection.MonoCMethod.Invoke (BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
  at System.Activator.CreateInstance (System.Type type, BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes) [0x00000] in <filename unknown>:0 
  at System.Activator.CreateInstance (System.Type type, System.Object[] args, System.Object[] activationAttributes) [0x00000] in <filename unknown>:0 
  at System.Activator.CreateInstance (System.Type type, System.Object[] args) [0x00000] in <filename unknown>:0 
  at GLib.Opaque.GetOpaque (IntPtr o, System.Type type, Boolean owned) [0x00000] in <filename unknown>:0 
  at Gst.Bus.TimedPopFiltered (UInt64 timeout, MessageType types) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Lib.Sound.Playback.GstreamerSharp.CGstreamerSharpAudioStream.Open (Boolean prescan) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Lib.Sound.Playback.CPlaybackBase.Load (System.String medium, Boolean loop, Boolean prescan) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CSound.Load (System.String media, Boolean loop, Boolean prescan) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CBsound.Load (System.String soundFile, Boolean loop, Boolean prescan) [0x00000] in <filename unknown>:0 
  at VocaluxeLib.Utils.Player.CSoundPlayer.Load (System.String file, Single position, Boolean autoplay) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CBackgroundMusic._StartSong () [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CBackgroundMusic.Next () [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CBackgroundMusic.Play () [0x00000] in <filename unknown>:0 
  at Vocaluxe.Screens.CScreenLoad.OnShowFinish () [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CGraphics.InitFirstScreen () [0x00000] in <filename unknown>:0 
  at Vocaluxe.Base.CDraw.MainLoop () [0x00000] in <filename unknown>:0 
  at Vocaluxe.CMainProgram._Run (System.String[] args) [0x00000] in <filename unknown>:0 
  at Vocaluxe.CMainProgram.Main (System.String[] args) [0x00000] in <filename unknown>:0

**Comment by derivator** [comment]
Unfortunately, my [previous fix](https://gist.github.com/derivator/fa0bdf9995e825c3a9fb) wasn't such a good idea afterall. I thought we could just use gstreamer-sharp.dll from the repository instead of requiring it to be installed, but I didn't know that gstreamer-sharp has a native component, `libgstreamersharpglue-1.0.0.so`, so it will have to be installed anyway. Unfortunately, that means you'll have to build gstreamer-sharp afterall, since your glue library seems to be outdated.

**Comment by AlexanderRavenheart** [comment]
Well, in the meantime I was working on that alternative, trying to compile gstreamer-sharp and I got to the point that I managed to build gtk-sharp3 which provides glib-sharp3.
I can't seem to get gstreamer1.0 updated to >=1.4.0 even with the ppa provided in an earlier comment.

I might try installing a different distribution with newer gstreamer packages and retrace the steps to see if I can get a working build.

**Comment by derivator** [comment]
That would probably work, although a method to get it working on Ubuntu would certainly be nice, since it's such a popular distribution.

**Comment by AlexanderRavenheart** [comment]
Yes, but it would probably be less of a hassle on a 14.10 or 15.04 version since on 14.04 it is so difficult to acquire the necessary versions of packages.

**Comment by AlexanderRavenheart** [comment]
If I get it working on something like OpenSuse or Fedora which are kind of the middle ground between bleeding-edge Arch and rarely updated Ubuntu, then I would try to get it working on a virtualized Ubuntu, but definitely not the LTS version.

**Comment by stsundermann** [comment]
It *should* work on 14.04 when installing from source because that's what I had when I worked on gstreamer-sharp. Compiling takes some time since you will need gstreamer, -base, -good, -bad and -libav

We *could* provide static compiled versions but that's not how packaging works. The best would be to have gtk# 2.99.4 and gstreamer-sharp 1.39.1 in the official packages but that's not going to happen before 15.04 or maybe even 15.10 (The background is that gstreamer-sharp 1.39.1 depends on gtk-sharp 2.99.4 which is yet unreleased so I cannot release gst#)

**Comment by AlexanderRavenheart** [comment]
The problem with the ppa is that it does not see the update for the plugins-bad. It remains at version 1.2.4 from trusty-updates while the other gstreamer packages have been updated to 1.4.3 from the ppa.

**Comment by stsundermann** [comment]
Well, in that case I'd recommend to update -bad from source. Grab the source for 1.4.3 from here http://gstreamer.freedesktop.org/src/gst-plugins-bad/ and do the usual ./configure --prefix=/usr && make && sudo make install

**Comment by AlexanderRavenheart** [comment]
Successfully compiled -bad from source and then successfully compiled gstreamer-sharp.
If I run the executable through Wine either by right clicking on it and selecting Wine or executing ./Vocaluxe.exe on the terminal, the same errors as mentioned here: https://github.com/Vocaluxe/Vocaluxe/issues/31#issuecomment-76546563 appear.

If I run the executable by right clicking on it and selecting Mono Runtime, the following error appears:
1) Error on start up: Vocaluxe.CLoadingException: Failed to load Language
  at Vocaluxe.CMainProgram._Run (System.String[] args) [0x00000] in <filename unknown>:0 
![screenshot - 01 03 2015 - 12 11 52](https://cloud.githubusercontent.com/assets/7274337/6430289/2d94e2b2-c00c-11e4-8c9a-0b5c4db513bb.png)

**Comment by stsundermann** [comment]
You don't need wine for Vocaluxe as it runs natively using mono. If you want to start from terminal you can use mono Vocaluxe.exe

Did you remove the previous gstreamer-sharp 0.99.0 before installing 1.39.1? Please check that /usr/lib and /usr/local/lib only contain libgstreamerglue-1.0.0.so from 1.39.1 (maybe determine through timestamps?)

**Comment by AlexanderRavenheart** [comment]
/usr/local/lib contains an older version. How do I update it?

**Comment by stsundermann** [comment]
You can do two things. Either run ./autogen.sh without specifying --prefix so that the version in /usr/local gets overridden or downgrade your repository to 0.99 run ./autogen.sh and run make uninstall

**Comment by AlexanderRavenheart** [comment]
Running without --prefix did not override the /usr/local/lib files.
I do not know how to downgrade the repository.

**Comment by stsundermann** [comment]
Maybe just clean it manually using sudo rm /usr/local/lib/libgstreamersharpglue* since you do have the right version in /usr/lib

**Comment by AlexanderRavenheart** [comment]
It updated properly. 
I did a make clean and make again in Vocaluxe before trying to run it again.
And ... it works :-D

**Comment by stsundermann** [comment]
Congrats!

**Comment by AlexanderRavenheart** [comment]
Well... now that we got it working... can we determine a set of clear steps that one would take to get it up and running from a fresh install of Ubuntu 14.04?

**Comment by derivator** [comment]
That's great! If there's any more problems/bugs, let us know.

@stsundermann is there an official location to download just the videos that aren't included in the repository, that we could point people who want to build Vocaluxe to, so they won't have to download a complete nightly? This would also help with creating an Arch Linux AUR package. ("Packages" distributed there aren't "ready to use"-archives, they're scripts that create a package by downloading from official mirrors or building from source)

**Comment by derivator** [comment]
@stsundermann Since Ubuntu 14.04 is going to be around for a while, it might be worth the effort to create an "official" PPA on launchpad (if I understand correctly, anyone can create one?) that has Vocaluxe and the needed dependencies (at least those that don't have their own official PPAs). What do you think, is it feasible, any objections?

**Comment by stsundermann** [comment]
@derivator I've attached the videos to the last tag on github https://github.com/Vocaluxe/Vocaluxe/releases/download/0.3.0.83/Videos.zip

An official ppa would be nice to have but we'll need a lot of dependencies (mono 3.12, gtk# 2.99.4, gstreamer 1.4 and gstreamer# 1.3.9.1) Everyone can create a ppa and you get 2GB of space for your packages

**Comment by derivator** [comment]
@stsundermann Thanks for taking care of the videos. I'll investigate creating a ppa in the next days...

**Comment by derivator** [comment]
@stsundermann this is a real nitpick, but your video zip has the videos in `Themes`, while Vocaluxe expects them in `Themes/Ambient`... For maximum drag-and-drop compatibility, it would be great if we could replicate that structure :)

**Comment by derivator** [comment]
>  this is a real nitpick, but your video zip has the videos in Themes, while Vocaluxe expects them in Themes/Ambient... For maximum drag-and-drop compatibility, it would be great if we could replicate that structure :)

@stsundermann could you confirm if the .zip is going to stay like it is or if you'll change it to reflect the folder structure that Vocaluxe expects? I'd like to change the HowToBuild documentation to include adding the videos, and this also affects my arch linux packaging script. Thanks and sorry for bothering you with this.

**Comment by stsundermann** [comment]
@derivator I've updated the package.

**Comment by lukeIam** [comment]
@derivator We could use a nuget package for deploying the files.
Here is a sample package: https://github.com/lukeIam/SpecialNugetPackages#lukeiamnugetdownloadandextract

Atm only MSBuild is supported but I plan to add xBuild compatibility (with a shell script).

**Comment by derivator** [comment]
@lukeIam I guess we could do that, but it would mean adding yet another dependency, right?

**Comment by lukeIam** [comment]
@derivator we would need a source feed for the package and build tools would re-add the package.

In my opinon it would only make sense if we use nuget for the other dependencies as well.
Atm getting and installing the dependencies for (windows) build is not very nice (And we lost our portable support as wen need to install gstreamer on all clients)
It raises the entry threshold for other beginners - we have many reports on the forum.

I'm thinking about a nuget solution for this problem:
My target is having a single Vocaluxe.Dependencies package which downloads all required dependencies as sub-packages.
The packages would detect the current platform and copy the corresponding files to the target folder.
(see https://github.com/lukeIam/SpecialNugetPackages#lukeiamnugetcopy)

The nuget packages would be supplied over a feed so there is only a kind of link in the project.
There are different possibilities for a feed:
- nuget.org (preinstalled in VS) 
     - package size limited to 30MB each
- myget.org (free plan for open source) 
    - package size limited to 100MB each 
    - feed must be added (could be done with a nuget.config in the project)
- own server (needs an ASP server as I know) 
    - packget size limit could be chosen freely 
    - feed must be added (could be done with a nuget.config in the project)

My next step would be to determinate which of all those gstreamer are really needed.

**Comment by Physkaj** [comment]
Hi there, I just found out your project existed and I am trying to run this using an Arch Linux system with the latest git (d967595) . It compiles (DebugLinux) but when I run it I have two issues:

1) My current directory must by either the install directory or the Output directory otherwise I get an error about Language.
2) When it does not complain about language i get the following error:

Unhandled Exception:
System.ObjectDisposedException: The object was used after being disposed.
  at Vocaluxe.Lib.Sound.Record.PitchTracker.CPtAKF._Dispose (Boolean disposing) [0x00000] in <filename unknown>:0 
  at Vocaluxe.Lib.Sound.Record.PitchTracker.CPitchTracker.Finalize () [0x00000] in <filename unknown>:0 

I am happy to try to assist in the testing of the Linux compatibility if you tell me what to do :)

Versions:
mono-4.0.1.28
gstreamer-sharp-1.4

**Comment by Nefrin** [comment]
@Physkaj the problem is that the pitch tracker, which causes your exception, is not written in cross-platform (Mono) C# but in Windows specific C/C++ (as far as I can tell). So if you wanted to compile it for linux you just would have to rewrite the pitch tracker for linux and replace it in the Vocaluxe source.

I personally got the program sort of starting on Mac OS X 10.9 and 10.10 by just downloading the v0.3.1 beta form [vocaluxe.de](http://vocaluxe.sourceforge.net/index.php?p=download) and start it with mono but it showed an error with the pitch tracker at the beginning which said it could not load because it was for the wrong platform. 

I believe if you changed the code to not use the pitch tracker you could get it to compile and start but you would not be able to sing because nothing tracks your voice :smile:

**Comment by Flamefire** [comment]
There shouldn't be anything windows specific in the pitchtracker and it is also unrelated to PortAudio. You only have to compile the pitchtracker for linux (maybe the project you loaded in linux did not do this?) and check, if vocaluxe finds the created library.
What is the "error about language"? What is it saying?
Are you able to debug the project?

**Comment by derivator** [comment]
@Physkaj please try with this [PKGBUILD](https://gist.github.com/derivator/585ea0013f4bd73f1faa). If a few people test and confirm that it works for them, I'll put it in the AUR. Not sure what causes your specific error, though I've seen that one before. Possibly missing ffmpeg?
@Nefrin the makefile builds the PitchTracker for linux, that's not the problem. Just running the windows build with mono will not work though, as you already discovered. There's another problem that sadly prevents a build for Mac OS X, I think it was some dependency not being available for that platform.

**Comment by Physkaj** [comment]
@Flamefire: I see that the pitchtracker is built during the build process and the corresponding libPitchTracker.dll.so is created in the Output directory. The language error message is "Error on start up: Failed to load Language". I don't know if it is related but I also have a message in the prompt telling me "Could not set X locale modifiers".

@derivator: I get exactly the same behavior when building using you pkgbuild )including language error). I forgot to mention it before, but I am able to see the splash screen (vocaluxe logo) before the error appears.

**Comment by derivator** [comment]
@Physkaj the language error shouldn't appear if you launch via the included desktop file. The working directory does indeed have to be the Output folder, but the desktop file should set this correctly. I'm not sure at the moment what causes the pitchtracker error. I'll investigate later, but unless I find something very quickly, it'll be next week before I can look at it again. If you're comfortable with that sort of stuff, you can try debugging in monodevelop to find the source of the problem.

**Comment by derivator** [comment]
@Physkaj are you on x86 or x64?

**Comment by Physkaj** [comment]
Ah, I'll check the desktop file out then, not running a desktop environment
here.

I'm on x64. I'll see if I can find something out, but I'm totally
unfamiliar with .NET and mono.

**Comment by derivator** [comment]
@Physkaj I'll make sure to include a script for people without desktop environments in upcoming versions... just tried building with monodevelop, for some reason that's a mess at the moment (complaints of missing assemblies, blah...) Unfortunately I won't be able to investigate further until next week. If I recall correctly, that error is caused by some problem with loading the dll, maybe the native code is crashing because some dependency is missing? I have
gst-libav 1.4.5-1
gst-plugins-bad 1.4.5-2
gst-plugins-base 1.4.5-1
gst-plugins-base-libs 1.4.5-1
gst-plugins-good 1.4.5-2
gstreamer 1.4.5-1
gstreamer-sharp 1.4-2
ffmpeg 1:2.6.3-1
ffmpeg-compat 1:0.10.16-1

installed.

**Comment by Physkaj** [comment]
@derivator: I have the same package versions as you do. I ran with MONO_LOG_LEVEL=debug and got lots of information. The relevant part seems to be that the .so file is found but it complains about an undefined symbol to standard out.

Mono: DllImport error loading library '/usr/share/vocaluxe/libPitchTracker.dll.so': '/usr/share/vocaluxe/libPitchTracker.dll.so: undefined symbol: _ZSt4cout'.

previous to this there is some problems loading libc, I don't understand if the library gets loaded or not in the end though (libc.so.6 is located in /usr/lib as it should be).

Mono: DllImport attempting to load: 'libc.so.6'.
Mono: DllImport error loading library '/usr/lib/mono/gac/Mono.Posix/4.0.0.0__0738eb9f132ed756/libc.so.6': '/usr/lib/mono/gac/Mono.Posix/4.0.0.0__0738eb9f132ed756/libc.so.6: cannot open shared object file: No such file or directory'.
Mono: DllImport error loading library '/usr/lib/mono/gac/Mono.Posix/4.0.0.0__0738eb9f132ed756/libc.so.6.so': '/usr/lib/mono/gac/Mono.Posix/4.0.0.0__0738eb9f132ed756/libc.so.6.so: cannot open shared object file: No such file or directory'.
Mono: DllImport loaded library 'libc.so.6'.
Mono: DllImport searching in: 'libc.so.6' ('libc.so.6').

I can provide the full log but it it very long and it seems I cannot attach it to this post.

**Comment by derivator** [comment]
@Physkaj libc error seems to be normal, as you can see from the last two lines, mono does eventually find it, and I get the same error.

"undefined symbol: _ZSt4cout" seems to be the culprit here. No idea what's causing it, yet, and I can't reproduce that output. Since we're running the same mono version, maybe it's a problem with the native compilation? Some incompatible header files? Different gcc versions? I don't have anything nonstandard installed though, I think. Will have to investigate next week.

**Comment by stsundermann** [comment]
According to the name the function is likely coming from C++'s cout. It might be a linking error

**Comment by Vittelius** [comment]
Hi, I tried to compile Vocaluxe with Ubuntu 15.04. I managed to fix all the linking-errors, but I still get this:
---------------------------------------------
make -C PitchTracker
make[1]: Entering directory '~/git/Vocaluxe/PitchTracker'
gcc -shared -o libPitchTracker.dll.so -fPIC compatibility.o dywapitchtrack/dywapitchtrack.o dywapitchtrack/ptDyWa.o FFT/FFT.o FFT/RealFFTf.o Helper.o performous/pitch.o ptAKF.o PitchWrapper.o
strip libPitchTracker.dll.so
cp libPitchTracker.dll.so ../Output/
make[1]: Leaving directory '~/git/Vocaluxe/PitchTracker'
make -C Vocaluxe/Lib/Video/Acinerella
make[1]: Entering directory '~/git/Vocaluxe/Vocaluxe/Lib/Video/Acinerella'
gcc -c -fPIC acinerella.c -I /usr/local/include
acinerella.c: In function ‘ac_get_stream_info’:
acinerella.c:201:67: error: ‘AVStream’ has no member named ‘r_frame_rate’
         (double)((lp_ac_data)pacInstance)->pFormatCtx->streams[nb]->r_frame_rate.num /
                                                                   ^
acinerella.c:202:67: error: ‘AVStream’ has no member named ‘r_frame_rate’
         (double)((lp_ac_data)pacInstance)->pFormatCtx->streams[nb]->r_frame_rate.den;
                                                                   ^
acinerella.c: In function ‘ac_free_package’:
acinerella.c:256:7: warning: ‘destruct’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:994) [-Wdeprecated-declarations]
       if (pkt->destruct){pkt->destruct = av_destruct_packet; pkt->destruct(pkt);} //FIXME: workaround for deprecation of AVPacket::destruct
       ^
acinerella.c:256:7: warning: ‘destruct’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:994) [-Wdeprecated-declarations]
acinerella.c:256:7: warning: ‘av_destruct_packet’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:3294) [-Wdeprecated-declarations]
acinerella.c:256:7: warning: ‘destruct’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:994) [-Wdeprecated-declarations]
acinerella.c: In function ‘ac_create_video_decoder’:
acinerella.c:309:2: warning: ‘avcodec_alloc_frame’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:3195) [-Wdeprecated-declarations]
  pDecoder->pFrame = avcodec_alloc_frame();
  ^
acinerella.c:310:2: warning: ‘avcodec_alloc_frame’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:3195) [-Wdeprecated-declarations]
  pDecoder->pFrameRGB = avcodec_alloc_frame();
  ^
acinerella.c: In function ‘ac_decode_audio_package’:
acinerella.c:467:3: warning: ‘avcodec_alloc_frame’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:3195) [-Wdeprecated-declarations]
   if (!(decoded_frame = avcodec_alloc_frame())) {
   ^
acinerella.c:471:9: warning: ‘avcodec_get_frame_defaults’ is deprecated (declared at /usr/include/libavcodec/avcodec.h:3205) [-Wdeprecated-declarations]
         avcodec_get_frame_defaults(decoded_frame);
         ^
acinerella.c:500:52: error: ‘AV_CH_LAYOUT_STEREO’ undeclared (first use in this function)
   av_opt_set_int(swr_ctx, "out_channel_layout",    AV_CH_LAYOUT_STEREO, 0);
                                                    ^
acinerella.c:500:52: note: each undeclared identifier is reported only once for each function it appears in
acinerella.c:511:84: error: ‘AV_ROUND_UP’ undeclared (first use in this function)
    decoded_frame->nb_samples, dst_sample_rate, (pDecoder->pCodecCtx)->sample_rate, AV_ROUND_UP);
                                                                                    ^
makefile:2: recipe for target 'acinerella' failed
make[1]: *** [acinerella] Error 1
make[1]: Leaving directory '~/Vocaluxe/Vocaluxe/Lib/Video/Acinerella'
makefile:4: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by derivator** [comment]
@Vittelius I strongly suspect this is because Ubuntu ships with libav instead of ffmpeg. For example, the [libav people helpfully decided that r_frame_rate is useless and should be removed.](https://git.libav.org/?p=libav.git;a=blobdiff;f=libavformat/avformat.h;h=067a7874b9593d0185ecf587d8cc99c7b648abb5;hp=08d2f3a8761fd670e5152c1e4488e0f1621ad748;hb=85a5bc054c1290699ccbf5799ba6c4e2fbcc3530;hpb=7b486ab13bfcfa88a7cd92586de50e49966ec292) I think you would need the package [ffmpeg](https://launchpad.net/ubuntu/+source/ffmpeg/7:2.5.7-0ubuntu0.15.04.1) and some/all of the lib*-ffmpeg-dev packages. I don't know if just installing them is sufficient, I think they may have their header files in different locations to avoid conflicts with libav, so you may have to change the gcc options in the makefile to `-lavformat-ffmpeg`, etc. Try how far you can get, and if you have any problems, post here and I'll try to help.

**Comment by lukeIam** [comment]
As the current devlop branch now supports Mono/Linux I'll close this issue.

I've moved your dependency problem to a new issue: #162

If there are other problems with the Linux support please open a new issue.

**Comment by lukeIam** [comment]
I thought that the real ffmpeg has returned in the official repo with Ubuntu 15.04
https://trac.ffmpeg.org/wiki/CompilationGuide/Ubuntu
